src/PythonMigrationViewExtension/GraphPythonDependencies.cs

```diff
@@ -46,14 +47,32 @@ internal bool CustomNodesContainIronPythonDependencies(IEnumerable<Function> cus
             {
                 customNodeManager.TryGetFunctionWorkspace(customNode.FunctionSignature, false, out customNodeWS);
 
-                if (customNodeWS.Nodes.Any(n => IsIronPythonNode(n)))
+                // If a custom node workspace is already checked for Python dependencies, 
+                // check the dictionary object instead of processing it again. 
+                if (CustomNodePythonDependency.ContainsKey(customNodeWS.CustomNodeId))
```

Comment:
You can use `TryGetValue` on the dictionary. You can check for the key and get the value for that key in one step.

Reply:
How do you keep the dictionary up-to-date if a custom node WS is edited - say, one or more python nodes are deleted and the WS no longer has any python node.

Reply:
I was also wondering this - but - since these messages are only raised once per session anyway maybe this is a non issue, next time dynamo is opened the data will be up to date.

Reply:
Yeah thats a good point, but like Mike said, it wouldn't be a problem if the python node is deleted from CN workspace because the dialog would be shown already when it was opened for the first time. On the other hand, when you add a new Python node to a CN workspace, save it and open that CN workspace again, the dialog would be shown for the first time because it checks the current workspace (that is opened) for Python dependencies here https://github.com/DynamoDS/Dynamo/pull/10777/files#diff-f13147450a5b0038bab03d9f6493aff8R32